src/consensus.rs

```diff
@@ -94,22 +86,20 @@ impl Consensus {
             channel_service,
             runtime.clone(),
         )?;
-        let shutdown_requested = consensus.shutdown_requested.clone();
 
         let state_ref_clone = state_ref.clone();
-        let consensus_handle =
-            thread::Builder::new()
-                .name("consensus".to_string())
-                .spawn(move || {
-                    if let Err(err) = consensus.start() {
-                        log::error!("Consensus stopped with error: {err}");
-                        state_ref_clone.on_consensus_thread_err(err);
-                    } else {
-                        log::info!("Consensus stopped");
-                        state_ref_clone.on_consensus_stopped();
-                    }
-                    Ok(())
-                })?;
+        thread::Builder::new()
+            .name("consensus".to_string())
+            .spawn(move || {
+                if let Err(err) = consensus.start() {
+                    log::error!("Consensus stopped with error: {err}");
+                    state_ref_clone.on_consensus_thread_err(err);
+                } else {
+                    log::info!("Consensus stopped");
+                    state_ref_clone.on_consensus_stopped();
+                    state_ref_clone.on_consensus_stopped();
```

Comment:
We should probably remove this line, because it is a duplicate.

Reply:
Was just a second too late. I'll open a separate PR for this right now.

Reply:
https://github.com/qdrant/qdrant/pull/2192

Reply:
will submit as another PR